homeassistant/components/qbittorrent/sensor.py

```diff
@@ -15,111 +17,211 @@
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import STATE_IDLE, UnitOfDataRate
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.typing import StateType
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
-from .const import DOMAIN
+from .const import (
+    DOMAIN,
+    STATE_ATTR_TORRENT_INFO,
+    STATE_DOWNLOADING,
+    STATE_SEEDING,
+    STATE_UP_DOWN,
+)
 from .coordinator import QBittorrentDataCoordinator
 
 _LOGGER = logging.getLogger(__name__)
 
-SENSOR_TYPE_CURRENT_STATUS = "current_status"
-SENSOR_TYPE_DOWNLOAD_SPEED = "download_speed"
-SENSOR_TYPE_UPLOAD_SPEED = "upload_speed"
-
-
-@dataclass
-class QBittorrentMixin:
-    """Mixin for required keys."""
-
-    value_fn: Callable[[dict[str, Any]], StateType]
-
-
-@dataclass
-class QBittorrentSensorEntityDescription(SensorEntityDescription, QBittorrentMixin):
```

Comment:
Having an entity description with value_fn is a best practice which qbittorrent was using and transmission recently also moved to this approach. I think this integration could still use this approach.

Reply:
Oh wow, this was changed like 2 days on the transmission integration! I'm gonna get some sleep, but I'll look into this new approach and update this PR accordingly. Thanks!

Reply:
Haha yep, and in that PR I also mentioned that the integration should use the service responses in the future.

Reply:
I don't see your comment on that, but I believe you :) Will look at it and make the required changes. Thanks!!